yb-voyager/src/srcdb/yugabytedb.go

```diff
@@ -0,0 +1,409 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package srcdb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/datafile"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils/sqlname"
+)
+
+type YugabyteDB struct {
+	source *Source
+
+	db *pgx.Conn
+}
+
+func newYugabyteDB(s *Source) *YugabyteDB {
+	return &YugabyteDB{source: s}
+}
+
+func (yb *YugabyteDB) Connect() error {
+	db, err := pgx.Connect(context.Background(), yb.getConnectionUri())
+	yb.db = db
+	return err
+}
+
+func (yb *YugabyteDB) Disconnect() {
+	if yb.db == nil {
+		log.Infof("No connection to the source database to close")
+		return
+	}
+
+	err := yb.db.Close(context.Background())
+	if err != nil {
+		log.Infof("Failed to close connection to the source database: %s", err)
+	}
+}
+
+func (yb *YugabyteDB) CheckRequiredToolsAreInstalled() {
+	checkTools("strings")
+}
+
+func (yb *YugabyteDB) GetTableRowCount(tableName string) int64 {
```

Comment:
This and many other methods are applicable only for offline migration. Do we need to implement them for YB?

Reply:
No, should I remove them or leave it for now?